pkg/dns-server/resolver/ip.go

```diff
@@ -0,0 +1,48 @@
+package resolver
+
+import (
+	"net"
+
+	"github.com/Nordix/simple-ipam/pkg/ipam"
+)
+
+type IPAM interface {
+	AllocateIP() (string, error)
+	FreeIP(ip string) error
+}
+
+type SimpleIPAM struct {
+	ipam.IPAM
+}
+
+func NewSimpleIPAM(cidr string) IPAM {
+	newIPAM, err := ipam.New(cidr)
+	if err != nil {
+		panic(err)
+	}
+
+	return &SimpleIPAM{*newIPAM}
+}
+
+func (i *SimpleIPAM) AllocateIP() (string, error) {
+	ip, err := i.Allocate()
+	if err != nil {
+		return "", err
+	}
+
+	return ip.String(), nil
+}
+
+func (i *SimpleIPAM) FreeIP(ip string) error {
+	parsedIP := net.ParseIP(ip)
+
+	// ensure the IP is reserved before deleting it
+	err := i.Reserve(parsedIP)
```

Comment:
Why is it important to reserve before `Free`?

Reply:
This a preacution, because free calls `delete` and it will panic in case there is no such IP.

Reply:
But from docs of that lib:
```go
// Free frees an allocated address.
// To free a non-allocated address is a no-op.
func (i *IPAM) Free(a net.IP) {
	delete(i.allocated, ipaddr.IPToIPv6Int(a))
}
```
And `delete` doesn't panic if there is no key